### ISSUE TITLE
recurring event exceptions

### ISSUE THREAD

**Issue by nbcnc** [fullcalendar/fullcalendar#4549]
https://fullcalendar.io/docs/v4/recurring-events

Would love an ability to provide exceptions to the recurring event.  In other words, My Event occurs weekly between Friday March 8th and Friday March 29th **except** for March 15th.  Exceptions could be an array that lets us provide multiple dates.

```
{
daysOfWeek:  ['5'],
startTime: '12:00',
endTime: '13:00',
startRecur: '2019-03-08',
endRecur: '2019-03-30',
exceptions: ['2019-03-15']
}
```

**Comment by flamber** [comment]
Hi @nbcnc 

Exclusions are supported in the `rrule` plugin: https://fullcalendar.io/docs/v4/rrule-plugin
```
{
  title: 'rrule string with exclusion',
  rrule: 'DTSTART:20190308T120000Z\nRRULE:FREQ=WEEKLY;UNTIL=20190330\nEXDATE:20190315T120000Z',
  duration: '01:00'
}
```
https://codepen.io/anon/pen/qvPxMZ?editors=0010

But it's only possible using the string-style - not the object-style - since the FullCalendar rrule plugin doesn't read the `EXDATE` or `EXRULE` parameters. (At least as of 4.0.0-beta.4)
Read more about the parameters (and other advanced stuff) on: https://github.com/jakubroztocil/rrule

**Comment by arshaw** [comment]
thanks. we should add support for exdate/exrule as properties of the event object somehow

**Comment by nbcnc** [comment]
Thanks @arshaw and @flamber.  I didn't notice that EXDATE was an option.  Adding it to the object style notation would definitely be helpful.  Here's an example of using EXDATE rule as a string in case anyone needs an example.

DTSTART:20190312T010000\nEXDATE:20190323T010000\nEXDATE:20190322T010000\nRRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=MO,TU,WE,TH,FR,SA,SU;UNTIL=20190330

Recurring starts on **2019-03-12 1AM** (MO-SU) until **2019-03-30** except **2019-03-22** and **2019-03-23**

**Comment by Blackbaud-SeanStephens** [comment]
While looking into this topic, I noticed that EXRULE is considered deprecated by the RFC standard, so support for it in future versions of RRule may be dropped. The standard prefers the use of EXDATE only.

I only mention in case it influences the implementation decision.

Looking forward to this one!

**Comment by ganySA** [comment]
EXRULE and EXDATE no longer work with the latest version of RRULE when using the parse string function.
RRULE says EXDATE is no longer a valid property.

**Comment by ganySA** [comment]
Any chance we can get gongAlls pull request merged as this feature no longer works under 5.

**Comment by ganySA** [comment]
Bump.

**Comment by arshaw** [comment]
Event recurrence exceptions have been implemented in v5.5.0 (as part of the `@fullcalendar/rrule` plugin)

Please look at the updated docs: https://fullcalendar.io/docs/rrule-plugin
